ddtrace/tracer/span.go

```diff
@@ -169,6 +170,37 @@ func (s *span) setSamplingPriority(priority int, sampler samplernames.SamplerNam
 	s.setSamplingPriorityLocked(priority, sampler, rate)
 }
 
+// setUser sets the span user ID tag as well as some optional user monitoring tags depending on the configuration.
+// The function assumes that the span it is called on is the trace's root span.
+func (s *span) setUser(id string, cfg UserMonitoringConfig) {
+	tags := map[string]string{}
+	keys := []string{keyUserEmail, keyUserName, keyUserScope, keyUserRole, keyUserSessionID}
+	vals := []string{cfg.email, cfg.name, cfg.scope, cfg.role, cfg.sessionID}
+	// Build optional tags map before locking
+	for i, v := range vals {
+		if v != "" {
+			tags[keys[i]] = v
+		}
```

Comment:
Why not replace all of this with a straight forward approach?
```suggestion
	tags := map[string]string{
		keyUserEmail: cfg.email,
		keyUserName: cfg.name,
		keyUserScope: cfg.scope,
		keyUserRole: cfg.role,
		keyUserSessionID: cfg.sessionID,
	}
```
And you're done :)

Worst case scenario, you may check if any of the value is `""` before setting the tag using `setMeta` in the `for` loop below. IMHO this turns out nicer but I am ok with your version too if you like it more.

Reply:
I wanted to avoid computation inside the lock but it's true the gain is neglectable. Also yes, your suggestion looks like cleaner Go code.